packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,316 @@
+  onChange,
+  selected,
+  className,
+  as = 'button',
+  ...rest
+}: TabsProps) {
+  const [activeTab, setActiveTab] = useState();
+  const [currentIndex, setCurrentIndex] = useState();
+  const [focusedState, setFocusedState] = useState([] as Array<string>);
+  const tabListRef = useRef<HTMLDivElement>(null);
+
+  const childrenArray = React.Children.toArray(children) as Array<
+    React.ReactElement
+  >;
+
+  useEffect(() => {
+    const currentIndex = childrenArray.reduce(
+      (acc, child, index) => {
+        if (!child) {
+          return acc;
+        }
+
+        if (selected || activeTab) {
+          return [activeTab, selected].includes(child.props.value)
+            ? [...acc, index]
+            : acc;
+        }
+
+        if (child.props.default) {
+          return [...acc, index];
+        }
+
+        return acc;
+      },
+      [] as Array<number>,
+    );
+
+    setCurrentIndex(currentIndex[0]);
+  });
+
+  function handleChange(e: React.SyntheticEvent<Element, MouseEvent>) {
+    if (!selected) {
+      setActiveTab((e.target as HTMLLIElement).getAttribute('data-tab-id'));
+    }
+
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  function handleKeyDown(e: React.KeyboardEvent) {
+    if (selected) {
+      return;
+    }
+
+    const enabledIndexes = childrenArray.reduce(
+      (acc, child, index) => {
+        if (child.props.disabled) {
+          return acc;
+        }
+
+        return [...acc, index];
+      },
+      [] as Array<number>,
+    );
+
+    const enabledCurrentIndex = enabledIndexes.indexOf(currentIndex);
+
+    let index: number;
+
+    switch (e.key) {
+      case 'ArrowRight':
+      case 'ArrowDown':
+        index = (enabledCurrentIndex + 1) % enabledIndexes.length;
+        setActiveTab(children[enabledIndexes[index]].props.value);
+        break;
+
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        index =
+          (enabledCurrentIndex - 1 + enabledIndexes.length) %
+          enabledIndexes.length;
+        setActiveTab(children[enabledIndexes[index]].props.value);
+        break;
+    }
+  }
+
+  function calcStyle() {
+    if (
+      !tabListRef ||
+      !tabListRef.current ||
+      typeof currentIndex !== 'number'
+    ) {
+      return css``;
+    }
+
+    const tabListChildren: Array<Element> = Array.from(
+      tabListRef.current.children,
+    );
+
+    let computedX = 0;
+
+    for (let i = 0; i < currentIndex || 0; i++) {
+      computedX += tabListChildren[i].scrollWidth;
+    }
+
+    return css`
+      transform: translate3d(${computedX}px, 0, 0);
+      width: ${tabListChildren[currentIndex].scrollWidth}px;
+    `;
+  }
+
+  // using an object here in order to alert consumer if duplicate values are used
+  const tabObject = childrenArray.reduce(
+    (acc, child, index) => {
+      if (!isTab(child)) {
+        return { ...acc, [index]: child };
+      }
+
+      const childValue = child.props.value;
+
+      if (Object.keys(acc).includes(childValue)) {
```

Comment:
```suggestion
      if (acc[childValue] != null) {
```

And then for the initial value for accumulator we can do:

```
{} as { [key: string]: React.ReactElement },
```

Reply:
👍